scripts/manage_demux_stats.py

```diff
@@ -0,0 +1,319 @@
+        if  found_undet > demux_process.udf["Maximum % Undetermined Reads per Lane"]:
+            #This looks kind of bad in lims, but \n can't be used.
+            problem_handler("warning", "Undemultiplexed reads for lane {} was {} ({})% thus exceeding \
+             defined limit.\t".format(lane_no, undet_lane_reads, found_undet))
+        else:
+            logger.info("Found {} ({}%) undemultiplexed reads for lane {}.".format(undet_lane_reads, found_undet, lane_no))
+
+"""Creates demux_{FCID}.csv and attaches it to process"""
+def write_demuxfile(proc_stats, demux_id):
+    #Includes windows drive letter support
+    datafolder = "{}_data".format(proc_stats["Instrument"])
+    lanebc_path = os.path.join(os.sep,"srv","mfs", datafolder,proc_stats["Run ID"],"laneBarcode.html")
+    try:
+        laneBC = classes.LaneBarcodeParser(lanebc_path)
+    except:
+        problem_handler("exit", "Unable to fetch laneBarcode.html from {}".format(lanebc_path))
+    fname = "{}_demuxstats_{}.csv".format(demux_id, proc_stats["Flow Cell ID"])
+    
+    #Writes less undetermined info than undemultiplex_index.py. May cause problems downstreams
+    with open(fname, "w") as csvfile:
+        writer = csv.writer(csvfile)
+        writer.writerow(["Project", "Sample ID", "Lane", "# Reads", "Index","Index name", "% of >= Q30 Bases (PF)"])
+        for entry in laneBC.sample_data:
+            index_name = ""
+            if "PF Clusters" in entry:
+                reads = entry["PF Clusters"]
+            else: 
+                reads = entry["Clusters"]
+                
+            if proc_stats["Paired"]:
+                reads = int(reads.replace(",",""))*2
+            else:
+                reads = int(reads.replace(",","")) 
+            
+            try:
+                writer.writerow([entry["Project"],entry["Sample"],entry["Lane"],reads, \
+                                 entry["Barcode sequence"],index_name,entry["% >= Q30bases"]])
+            except:
+                problem_handler("exit", "Flowcell parser is unable to fetch all necessary fields for demux file.")
+    return laneBC.sample_data
+
+@click.command()
+@click.option("--process_lims_id", required=True,help="Lims ID of process. Example:24-92373")
+@click.option("--demux_id", required=True,help="Id prefix for demux output.")
+@click.option("--log_id", required=True,help="Id prefix for logfile")
+def main(process_lims_id, demux_id, log_id ):
+    #Sets up logger
+    basic_name = "{}_logfile.txt".format(log_id)
+    logger.setLevel(logging.DEBUG)
+    fh = logging.FileHandler("{}_logfile.txt".format(log_id))
+    fh.setLevel(logging.DEBUG)
+    ch = logging.StreamHandler()
+    ch.setLevel(logging.DEBUG)
+    formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+    fh.setFormatter(formatter)
+    ch.setFormatter(formatter)
+    logger.addHandler(ch)
+    logger.addHandler(fh)
+
```

Comment:
Useful for command line debugging as well as providing some pretty good feedback as to why something has failed. Right now it's only used for exceeding "Maximum % Undetermined Reads per Lane", which isn't a hard threshold; just kind of.